server/activityItem/fetch.ts

```diff
@@ -277,12 +279,19 @@ const fetchAssociations = (
 	});
 };
 
+export const fetchAssociationsForActivityItems = async (
+	activityItems: types.ActivityItem[],
+	scope?: Scope,
+) => {
+	const associationIds = getActivityItemAssociationIds(activityItems, scope);
+	return fetchAssociations(associationIds);
+};
+
 export const fetchActivityItems = async (
 	options: FetchActivityItemsOptions,
 ): Promise<ActivityItemsFetchResult> => {
 	const { offset = 0, limit = 50, scope, filters = [] } = options;
 	const activityItems = await fetchActivityItemModels({ offset, limit, scope, filters });
-	const associationIds = getActivityItemAssociationIds(activityItems, options.scope);
-	const associations = await fetchAssociations(associationIds);
+	const associations = await fetchAssociationsForActivityItems(activityItems, options.scope);
```

Comment:
does this mean, instead of fetching all activity, you are fetching activity per association to a pub or collection

Reply:
On this line we've already fetched the `ActivityItems` and we're now requesting models associated with those items — for instance a `pub-discussion-created` activity item will want to load the associated `Discussion` model. We're just changing how the `scope` gets passed around because we now want to expose the `fetchAssociationsForActivityItems` function for callers that _already_ have activity items.